src/Compilers/VisualBasic/Portable/Binding/Binder_Query.vb

```diff
@@ -2117,7 +2116,8 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
                                                                               intoRangeVariables,
                                                                               diagnostics)
 
-            Dim intoLambda As New BoundQueryLambda(syntaxNode,
+            Dim syntax As VisualBasicSyntaxNode = aggregationVariables.FirstOrDefault()
+            Dim intoLambda As New BoundQueryLambda(If(syntax, syntaxNode),
```

Comment:
It doesn't feel right to associate this lambda with the first aggregation variable.